### PR TITLE
Fix Bug: the error message when the application backup is performed 

### PR DESCRIPTION
**Pull request by yangkaa**
- When the stateful component is not deployed, the backup application will give an error message that the component is not closed. Therefore, this error is ignored when the component has no deployment version.

### CODE REVIEW

File: api/handler/group/group_backup.go [hunk 1]
```diff
@@ -239,7 +239,14 @@ func (h *BackupHandle) snapshot(ids []string, sourceDir string, force bool) erro
 		status := h.statusCli.GetStatus(id)
 		logrus.Debugf("service: %s is state: %v", service.ServiceAlias, service.IsState())
 		if !force && status != v1.CLOSED && service.IsState() { // state running service force backup
-			return fmt.Errorf("state app must be closed before backup")
+			versions, err := db.GetManager().VersionInfoDao().GetVersionByServiceID(id)
```
Comment (barnettZQG) [inline api/handler/group/group_backup.go:242]: if the component does not have versions, the status is undepoy? so judgment base status?

File: api/handler/group/group_backup.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func (h *BackupHandle) snapshot(ids []string, sourceDir string, force bool) erro
 		}
 		status := h.statusCli.GetStatus(id)
 		logrus.Debugf("service: %s is state: %v", service.ServiceAlias, service.IsState())
-		if !force && status != v1.CLOSED && service.IsState() { // state running service force backup
+		if !force && status != v1.CLOSED && status != v1.UNDEPLOY && service.IsState() { // state running service force backup
```
Comment (barnettZQG) [inline api/handler/group/group_backup.go:241]: if the component status is UNDEPLOY, it can be backuped?

Comment (yangkaa) [reply api/handler/group/group_backup.go:241]: Undeployed components can be backed up

Comment (barnettZQG) [reply api/handler/group/group_backup.go:241]: Component backups that are not currently in version will fail.

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**